src/components/Todo.tsx

```diff
@@ -7,79 +7,113 @@ export interface Todo {
   id: number;
   title: string;
   isCompleted: boolean;
+  date: string;
 }
 
 export default function Todos() {
   const [todolist, setTodolist] = useState<Todo[]>([]);
   const [trigger, setTrigger] = useState(0);
-  const [showCompleted, setShowCompleted] = useState(false);
+  const [search, setSearch] = useState<string>("");
+  const [sortOrder, setSortOrder] = useState<"asc" | "desc">("asc");
+  const [statusFilter, setStatusFilter] = useState<boolean | null>(null);
 
   const navigate = useNavigate();
 
   useEffect(() => {
     axios.get("http://localhost:8000/todos")
-      .then((resp) => (setTodolist(resp.data)))
+      .then((resp) => setTodolist(resp.data))
       .catch((error) => console.error("Error fetching todos:", error));
   }, [trigger]);
 
-  const handleCheckbox = (
-    todo: Todo,
-    e: React.ChangeEvent<HTMLInputElement>
-  ) => {
+  const handleCheckbox = (todo: Todo, e: React.ChangeEvent<HTMLInputElement>) => {
     axios.patch(`http://localhost:8000/todos/${todo.id}`, {
       isCompleted: e.target.checked,
     })
-    .then(() => setTrigger(trigger => trigger + 1))
-    .catch((error) => console.error("Error updating todo:", error));
+      .then(() => setTrigger(trigger => trigger + 1))
```

Comment:
Any reason for this trigger state? Can we remove this as its not relevant to our application